polling/src/main/java/io/atleon/polling/AloPollingReceiver.java

```diff
@@ -0,0 +1,64 @@
+package io.atleon.polling;
+
+import io.atleon.core.Alo;
+import io.atleon.core.AloFactory;
+import io.atleon.core.AloFlux;
+import io.atleon.core.OrderManagingAcknowledgementOperator;
+import io.atleon.polling.reactive.Poller;
+import io.atleon.polling.reactive.PollerOptions;
+import io.atleon.polling.reactive.PollingReceiver;
+import io.atleon.polling.reactive.PollingSourceConfig;
+import io.atleon.polling.reactive.ReceiverRecord;
+import reactor.core.publisher.Flux;
+import reactor.core.publisher.Sinks;
+import reactor.core.scheduler.Schedulers;
+
+import java.util.function.Function;
+
+public class AloPollingReceiver<T> {
+
+    private final Poller<T> poller;
+    private final PollingSourceConfig config;
+
+    private AloPollingReceiver(final Poller<T> poller,
+                              final PollingSourceConfig config) {
+        this.poller = poller;
+        this.config = config;
+    }
+
+    public static <T> AloPollingReceiver<T> from(final Poller<T> poller,
+                                                 final PollingSourceConfig config) {
+        return new AloPollingReceiver<>(poller, config);
+    }
+
+    public AloFlux<T> receiveRecords() {
+        return Flux.defer(this::receiveAloRecords)
+                .as(AloFlux::wrap);
+    }
+
+    private Flux<Alo<T>> receiveAloRecords() {
+        final AloFactory<T> aloFactory = new AloPollingRecordFactory<>();
+        return PollingReceiver.create(poller, buildPollerOptions()).receive()
+                .transform(r -> createAloRecords(r, aloFactory));
+    }
+
+    private PollerOptions buildPollerOptions() {
+        return PollerOptions.create(config.getPollingInterval(),
+                () -> Schedulers.newSingle(AloPollingReceiver.class.getSimpleName()));
+    }
+
+    private Flux<Alo<T>> createAloRecords(final Flux<ReceiverRecord<T>> records,
+                                          final AloFactory<T> aloFactory) {
+        final Sinks.Empty<Alo<T>> sink = Sinks.empty();
+        return records.map(record -> aloFactory.create(record.getRecord(),
+                        () -> record.getPoller().ack(record.getRecord()),
+                        t -> {
+                            sink.tryEmitError(t);
+                            record.getPoller().nack(record.getRecord());
+                        }))
+                .mergeWith(sink.asMono())
+                .transform(aloRecords -> new OrderManagingAcknowledgementOperator<>(
+                        aloRecords,
+                        Function.identity()));
```

Comment:
I don't think you want to use `Function.identity()` here; This will result in no correlation between the emitted polled records, resulting in no order management. You likely want to just return a constant value here, or delegate to the Poller to extract the group for you (`Object extractGroup(T t);`)

Reply:
Good call, I left this in there by accident from testing. I'll get that updated.